lsp/nls/src/requests/completion.rs

```diff
@@ -47,7 +47,7 @@ impl From<Ident> for IdentWithType {
     fn from(ident: Ident) -> Self {
         IdentWithType {
             ident,
-            ty: Types(TypeF::Dyn),
+            ty: Types::with_default_pos(TypeF::Dyn),
```

Comment:
As a datapoint, we use `from` inside `RichTerm` for that. It doesn't make a huge difference, but it's shorter and one function less to implement: `Types::from(TypeF::Dyn)`. But I let the decision to you. It could be a `new` method as well.